frontends/web/src/connected.tsx

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright 2018 Shift Devices AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { h, RenderableProps } from 'preact';
+import { subscribe } from './decorators/subscribe';
+import { App } from './app';
+
+interface Props {
+    connected?: boolean;
+}
+
+function ConnectedApp({ connected = true }: RenderableProps<Props>): JSX.Element {
+    return connected ? <App /> : (
+        <div className="app" style="padding: 40px">
+            The WebSocket closed. Please restart the backend and reload this page.
```

Comment:
Is this for a dev-only version? Otherwise, I don't thing any user will understand what "websocket closed" and "restart the backedn" mean.

I'd leave it the way it was, "An error occurred. Please restart the application.".

Reply:
> Since the Qt app has no WebSocket, it never triggers the disconnected event. That's why I also made the error text a bit more specific.

Reply:
Sure, but what about non-qt app? You don't have any today, but about tomorrow? Tomorrow you'll probably need to go back and change it back. So, why not leave it as is for now and maybe change it later when you really need to?

Reply:
At that point one should internationalize the string anyway…

Reply:
Sure but it doesn't have to be in this PR. This one is titled "refactor backend connected state from app.jsx to own component", so why should you care and change the actual wording or add translation support.

Anyway, I'm not insisting. It was just a suggestion.

Reply:
Yes, I see this, thanks. I just thought it would improve the clarity of both the code and the error message when faced with it as a new developer. (I could also have done the former with a comment, I know.)